Userland/Libraries/LibRegex/RegexByteCode.cpp

```diff
@@ -701,21 +701,41 @@ const Vector<String> OpCode_Compare::variable_arguments_to_string(Optional<Match
 
     for (size_t i = 0; i < arguments_count(); ++i) {
         auto compare_type = (CharacterCompareType)m_bytecode->at(offset++);
-        result.empend(String::format("type=%lu [%s]", (size_t)compare_type, character_compare_type_name(compare_type)));
+        result.empend(String::formatted("type={} [{}]", (size_t)compare_type, character_compare_type_name(compare_type)));
 
         auto compared_against_string_start_offset = state().string_position > 0 ? state().string_position - 1 : state().string_position;
 
         if (compare_type == CharacterCompareType::Char) {
-            char ch = m_bytecode->at(offset++);
-            result.empend(String::format("value='%c'", ch));
-            if (!view.is_null() && view.length() > state().string_position)
-                result.empend(String::format(
-                    "compare against: '%s'",
-                    view.substring_view(compared_against_string_start_offset, state().string_position > view.length() ? 0 : 1).to_string().characters()));
+            auto ch = m_bytecode->at(offset++);
+            auto is_ascii = isascii(ch) && isprint(ch);
+            if (is_ascii)
+                result.empend(String::formatted("value='{:c}'", static_cast<char>(ch)));
+            else
+                result.empend(String::formatted("value={}", ch));
+
+            if (!view.is_null() && view.length() > state().string_position) {
+                if (is_ascii) {
+                    result.empend(String::formatted(
+                        "compare against: '{}'",
+                        view.substring_view(compared_against_string_start_offset, state().string_position > view.length() ? 0 : 1).to_string()));
+                } else {
+                    auto str = view.substring_view(compared_against_string_start_offset, state().string_position > view.length() ? 0 : 1).to_string();
+                    u64 value = 0;
+                    if (str.length() >= sizeof(u64))
+                        value = *reinterpret_cast<const u64*>(str.characters());
```

Comment:
Isn't this going to cause potentially unaligned access? Shouldn't this be done using `memcpy`?

Reply:
Honestly, it's in debug code...
I was thinking this as I wrote it, then I thought "well it's debug code, so who cares if it blows up"...

I'll fix it when I feel less lazy, draft until then I guess

Reply:
Hmmm, actually, why even bother, let's just dump out some hex instead.

Reply:
I didn't even realize it was debug code. Anyway, on x86 this works with a slight speed penalty, though we should probably enable the alignment check at some point, which would then also lead to a crash. But it's not portable and not all architectures allow unaligned access. Older ARMs for example didn't allow it.

Reply:
if we're debugging LibRegex on a non-x86 arch, I'll consider even the crash a win lol

Reply:
Well wait until @awesomekling throws in the alignment check bit and it'll crash on x86 as well ;)